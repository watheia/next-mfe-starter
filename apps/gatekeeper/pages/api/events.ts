import type { Endpoints } from '@octokit/types';
import { sessionOptions } from '@watheia/mfe.api';
import { withIronSessionApiRoute } from 'iron-session/next';
import { NextApiRequest, NextApiResponse } from 'next';
import { Octokit } from 'octokit';

export type Events = Endpoints['GET /users/{username}/events']['response']['data'];

const octokit = new Octokit();

export default withIronSessionApiRoute(handler, sessionOptions);

async function handler(req: NextApiRequest, res: NextApiResponse<Events>) {
  const user = req.session?.user;

  if (!user || !user.isLoggedIn) {
    res.status(401).end();
    return;
  }

  try {
    const { data: events } = await octokit.rest.activity.listPublicEventsForUser({
      username: user.login,
    });

    res.json(events);
  } catch (error) {
    res.status(200).json([]);
  }
}
