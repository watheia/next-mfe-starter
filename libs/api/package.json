{
  "name": "@watheia/mfe.api",
  "version": "0.0.1"
}
