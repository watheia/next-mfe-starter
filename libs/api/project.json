{
  "root": "libs/api",
  "sourceRoot": "libs/api/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/web:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/api",
        "tsConfig": "libs/api/tsconfig.lib.json",
        "project": "libs/api/package.json",
        "entryFile": "libs/api/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "compiler": "swc",
        "assets": [
          {
            "glob": "libs/api/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/api/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/api"],
      "options": {
        "jestConfig": "libs/api/jest.config.js",
        "passWithNoTests": true
      }
    }
  }
}
