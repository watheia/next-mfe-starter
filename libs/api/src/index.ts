export { AuthProvider, useAuth } from './auth-context';
export type { AuthProviderProps } from './auth-context';
export { default as env } from './env';
export { default as fetcher, FetchError } from './fetcher';
export { SupabaseContext, SupabaseProvider, useSupabase } from './supabase';
export type { SupabaseContextType, SupabaseProviderProps } from './supabase';
export type { WaAuthContext } from './types/WaAuthContext';
export type { WaSignInRequest } from './types/WaSignInRequest';
export type { WaSignInResponse } from './types/WaSignInResponse';
export type { WaSignOutResponse } from './types/WaSignOutResponse';
export type { WaSignUpRequest } from './types/WaSignUpRequest';
export type { WaSignUpResponse } from './types/WaSignUpResponse';
export type { WaUser } from './types/WaUser';
export { default as usePrincipal } from './use-principal';
export { default as url, isExternal } from './url';
