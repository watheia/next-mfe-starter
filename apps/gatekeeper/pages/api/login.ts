import type { WaUser } from '@watheia/mfe.api';
import { sessionOptions } from '@watheia/mfe.api';
import { withIronSessionApiRoute } from 'iron-session/next';
import { NextApiRequest, NextApiResponse } from 'next';
import { Octokit } from 'octokit';

const octokit = new Octokit();

export default withIronSessionApiRoute(handler, sessionOptions);

async function handler(req: NextApiRequest, res: NextApiResponse) {
  const { username } = await req.body;

  try {
    const {
      data: { login, avatar_url },
    } = await octokit.rest.users.getByUsername({ username });

    const user = { isLoggedIn: true, login, avatarUrl: avatar_url } as WaUser;
    req.session.user = user;
    await req.session.save();
    res.json(user);
  } catch (error) {
    res.status(500).json({ message: (error as Error).message });
  }
}
