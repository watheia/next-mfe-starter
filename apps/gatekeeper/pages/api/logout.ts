import type { WaUser } from '@watheia/mfe.api';
import { sessionOptions } from '@watheia/mfe.api';
import { withIronSessionApiRoute } from 'iron-session/next';
import { NextApiRequest, NextApiResponse } from 'next';

export default withIronSessionApiRoute(handler, sessionOptions);

function handler(req: NextApiRequest, res: NextApiResponse<WaUser>) {
  req.session.destroy();
  res.json({ isLoggedIn: false, login: '', avatarUrl: '' });
}
