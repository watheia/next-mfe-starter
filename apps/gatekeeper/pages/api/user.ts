import { sessionOptions } from '@watheia/mfe.api';
import { withIronSessionApiRoute } from 'iron-session/next';
import { NextApiRequest, NextApiResponse } from 'next';

export type User = {
  isLoggedIn: boolean;
  login: string;
  avatarUrl: string;
};

export default withIronSessionApiRoute(handler, sessionOptions);

async function handler(req: NextApiRequest, res: NextApiResponse<User>) {
  if (req.session.user) {
    // in a real world application you might read the user id from the session and then do a database request
    // to get more information on the user if needed
    res.json({
      ...req.session.user,
      isLoggedIn: true,
    });
  } else {
    res.json({
      isLoggedIn: false,
      login: '',
      avatarUrl: '',
    });
  }
}
