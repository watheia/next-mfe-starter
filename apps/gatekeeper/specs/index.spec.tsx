import { render } from '@testing-library/react';
import { AuthProvider, env, SupabaseProvider } from '@watheia/mfe.api';
import React from 'react';
import Index from '../pages/index';

describe('Index', () => {
  it('should render successfully', () => {
    const { container } = render(
      <SupabaseProvider publicUrl={env.supabase_url} anonKey={env.public_key}>
        <AuthProvider>
          <Index />
        </AuthProvider>
      </SupabaseProvider>
    );
    expect(container).toBeInstanceOf(HTMLElement);
    expect(container).toMatchSnapshot();
  });
});
