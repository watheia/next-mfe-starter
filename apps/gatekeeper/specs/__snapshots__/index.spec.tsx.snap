// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should render successfully 1`] = `
<div>
  <main
    class="MuiGrid-root MuiGrid-container css-1jrn2fe-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-4 MuiGrid-grid-md-7 css-f50ftj-MuiGrid-root"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation6 MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-5 css-gadse4-MuiPaper-root-MuiGrid-root"
    >
      <div
        class="MuiBox-root css-1y8ugea"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="LockOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
            />
          </svg>
        </div>
        <h1
          class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
        >
          Sign in
        </h1>
        <form
          class="MuiBox-root css-164r41r"
          novalidate=""
        >
          <div
            class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused Mui-required css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="email"
              id="email-label"
            >
              Email Address
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                autocomplete="email"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="email"
                name="email"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-1z7n62"
                >
                  <span>
                    Email Address
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Send token
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </main>
</div>
`;
